src/lib/utils.ts

```diff
@@ -0,0 +1,96 @@
+import { Readable, Writable } from 'stream';
+import toArray = require('stream-to-array');
+import tmp = require('tmp');
+
+/**
+ * Stream a readable stream into memory.
+ * @param stream
+ */
+export async function streamToBuffer(stream: Readable): Promise<Buffer> {
+    const streamParts = await toArray(stream);
+    const streamBuffers: Buffer[] = streamParts.map(
+        (part: ArrayBuffer | SharedArrayBuffer): Buffer => {
+            return Buffer.isBuffer(part) ? part : Buffer.from(part);
+        },
+    );
+    return Buffer.concat(streamBuffers);
+}
+
+/**
+ * Pipe a readable stream into a writable stream and resolve once it's completely piped or reject if the pipe fails.
+ *
+ * TODO Implement timeout.
```

Comment:
Do we need a task to cover implementing a timeout? Is this TODO still relevant?

Reply:
Yep still relevant but nowhere to track it. This isn't crucial to functionality and just a note to a future developer who wants to implement this feature.